Sources/Post.php

```diff
@@ -2340,10 +2340,6 @@ function Post2()
 			$msgOptions['poster_time'] = $row['poster_time'];
 		}
 
-		// This will save some time...
-		if (empty($approve_has_changed))
-			unset($msgOptions['approved']);
-
```

Comment:
I think this was here to stop unwanted approval state change.

https://github.com/SimpleMachines/SMF/blob/acee5b182b8f692c99fb4c6af08eb1d0258a4cfe/Sources/Subs-Post.php#L2392-L2394

Reply:
Hmmm...  I don't think I see how that can happen.  If the approval state didn't change, and it had been approved, it sure looks OK to invoke the approval logic without harm.  Tests OK, too.

Have you found a specific scenario that fails?  How can 'approved' get incorrectly set?

OTOH, if the approval state did not change, and the post had been previously approved, there is a definite issue here...  It needs to get to approvePosts().

Reply:
& BTW, thanks for taking a look...